sdk/keyvault/Azure.Security.KeyVault.Secrets/samples/Sample4_GetSecretIfExists.md

```diff
@@ -0,0 +1,55 @@
+# Get a secret without throwing if it is not defined
+
+This sample demonstrates how to get a secret without throwing an exception if it is not defined.
+This may be useful in some application configuration managers that attempt to fetch key/value pairs from a collection of providers and when exceptions may break startup or are otherwise costly.
+To better configure ASP.NET applications or other applications that use common [.NET Configuration], see our documentation on [Azure.Extensions.AspNetCore.Configuration.Secrets].
+To get started, you'll need a URI to an Azure Key Vault. See the [README](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Secrets/README.md) for links and instructions.
+
+## Creating a SecretClient
+
+To create a new `SecretClient` to create get a secret, you need the endpoint to an Azure Key Vault and credentials.
+You can use the [DefaultAzureCredential] to try a number of common authentication methods optimized for both running as a service and development.
+
+In the sample below, you can set `keyVaultUrl` based on an environment variable, configuration setting, or any way that works for your application.
+
+```C# Snippet:SecretsSample4SecretClient
+var client = new SecretClient(new Uri(keyVaultUrl), new DefaultAzureCredential());
+```
+
+## Get a secret
+
+To prevent throwing a `RequestFailedException` when a secret does not exist, or to alter other behaviors of a specific API call, you can create a `RequestContext` and pass that to the API call:
```

Comment:
Would it be worth linking to some of the docs on RequestContext here, since this may be a new concept for Azure SDK .NET customers?